src/aaudio/AudioStreamAAudio.cpp

```diff
@@ -298,6 +297,14 @@ Result AudioStreamAAudio::close() {
     // This will delete the AAudio stream object so we need to null out the pointer.
     AAudioStream *stream = mAAudioStream.exchange(nullptr);
     if (stream != nullptr) {
+        // Sometimes a callback can occur shortly after a stream has been stopped and
+        // even after a close. If the stream has been closed then the callback
+        // can access memory that has been freed. That causes a crash.
+        // Two milliseconds may be enough but 10 msec is even safer.
+        // This seems to be less likely in Q but we have seen some reports.
+        if (getSdkVersion() <= __ANDROID_API_Q__) {
+            usleep(kDelayBeforeCloseMillis * 1000);
```

Comment:
Clients could call close() even after the stream is closed. In those cases, the forced delay may be unnecessary. Would it help to check the state before enforcing the delay?

Reply:
If close () is called a second time then stream would be null and that code would be skipped.

Reply:
I see now. Thanks for the clarification.